### PR TITLE
[doc] Landing page: Added why RAG wins over fine-tuning

### PR DESCRIPTION
**Pull request by tslmy**
# Description

The intuitive solution to "LLM doesn't know my data", unfortunately for many, is still "let's fine-tune a LLM".

Here I'm trying to convince those people that "RAG is all you need, without trade-offs".

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Added new unit/integration tests
- [ ] Added new notebook (that tests end-to-end)
- [ ] I stared at the code and made sure it makes sense

# Suggested Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added Google Colab support for the newly added notebooks.
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

File: docs/index.rst [hunk 1]
```diff
@@ -10,7 +10,25 @@ LLMs offer a natural language interface between humans and data. Widely availabl
...
 However, while LLMs are trained on a great deal of data, they are not trained on **your** data, which may be private or specific to the problem you're trying to solve. It's behind APIs, in SQL databases, or trapped in PDFs and slide decks.
 
-LlamaIndex solves this problem by connecting to these data sources and adding your data to the data LLMs already have. This is often called Retrieval-Augmented Generation (RAG). RAG enables you to use LLMs to query your data, transform it, and generate new insights. You can ask questions about your data, create chatbots, build semi-autonomous agents, and more. To learn more, check out our Use Cases on the left.
+You may choose to **fine-tune** a LLM with your data, but:
+
+- Training a LLM is **expensive**.
+- Due to the cost to train, it's **hard to update** a LLM with latest information.
+- **Observability** is lacking. When you ask a LLM a question, it's not obvious how the LLM arrived at its answer.
+
+LlamaIndex takes a different approach called `Retrieval-Augmented Generation (RAG) <https://research.ibm.com/blog/retrieval-augmented-generation-RAG>`_. Instead of asking LLM to generate an answer immediately, LlamaIndex:
```
Comment (hatianzhang) [inline docs/index.rst:19]: not sure whether we should use this link for RAG definition.

Comment (tslmy) [reply docs/index.rst:19]: Yeah, it was an arbitrary choice. That's the top result when I searched on Google. Feel free to add suggestion using that "add suggestion" button for a better link, if you have one in mind.

Comment (logan-markewich) [reply docs/index.rst:19]: Lets link to our own definition, in the context of llama-index. Just pushed a change

**Review by logan-markewich — APPROVED** [review]
I like it, thanks for the change!

**Pull request merged**